include/gsl/gsl

```diff
@@ -117,6 +117,42 @@ std::ostream& operator<<(std::ostream& os, const not_null<T>& val)
     return os;
 }
 
+template <class T, class U>
+bool operator==(const not_null<T>& lhs, const not_null<U>& rhs)
+{
+    return lhs.get() == rhs.get();
+}
```

Comment:
I agree with @CaseyCarter, it would probably be best to SFINAE here (roll on Concepts, sigh). You can just push more changes to this PR if you like.